rero/rero-ils
# cypress: improve tests

* Removes useless code.
* Replaces UI actions by API calls where possible in order to save time
during the tests.
* Replaces UI navigation by cy.visit instruction when it's relevant
for the test.
* Adds a method to get current date and hour in order to use it in the
API requests.
* Adds methods to create a document and an item with API call.
* Updates fixtures for the requests body.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>

## Why are you opening this PR?

To improve and fix cypress tests (they will need less time).

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui v0.14 branch
* dev of ng-core

## How to test?

You may need a bootstrap and a setup, depending on your local state.
You must have the correct branchs of rero-ils-ui and ng-core locally (see dependencies above).
`poetry run folding -c path/to/rero/ng-core -u path/to/rero/rero-ils-ui`
`FLASK_DEBUG=False poetry run server`
`poetry run cypress`

## Code review check list
